File: Arrays/P015.py
def brute(nums,target):
    n=len(nums)
    c=0
    for i in range(n):
        s=0
        for j in range(i+1,n):
            s+=nums[j]
        if s==target:
            c+=1
    return c

def optimal(nums,target):
    n=len(nums)
    c=0
    prev_sum=set()
    s=0
    for n in nums:
        s+=n
        if s==target:
            c+=1
        elif s>target:
            if s-target in prev_sum:
                c+=1
        prev_sum.add(s)
    return c

if __name__=='__main__':
    nums=input('Enter numbers: ').split()
    nums=list(map(int,nums))
    target=int(input('Enter target: '))
    print('Brute: ',brute(nums,target))
    print('optimal: ',optimal(nums,target))

File: Striver_C++_Problems/Arrays/P015.cpp
// P015.cpp - Count subarrays with sum == target

#include <bits/stdc++.h>
using namespace std;

int brute(const vector<int>& nums, int target) {
    int n = nums.size();
    int c = 0;
    for (int i = 0; i < n; i++) {
        int s = 0;
        for (int j = i + 1; j < n; j++) {   // mirrors your Python
            s += nums[j];
        }
        if (s == target) c++;
    }
    return c;
}

int optimal(const vector<int>& nums, int target) {
    int n = nums.size();
    int c = 0;
    unordered_set<long long> prev_sum;
    long long s = 0;
    for (int v : nums) {
        s += v;
        if (s == target) c++;
        else if (s > target && prev_sum.find(s - target) != prev_sum.end())
            c++;
        prev_sum.insert(s);
    }
    return c;
}

int main() {
    cout << "Enter numbers: ";
    string line;
    getline(cin, line);
    stringstream ss(line);
    vector<int> nums;
    int x;
    while (ss >> x) nums.push_back(x);

    cout << "Enter target: ";
    int target;
    cin >> target;

    cout << "Brute: " << brute(nums, target) << "\n";
    cout << "optimal: " << optimal(nums, target) << "\n";
    return 0;
}
